### PR TITLE
Implement .count(), .first(), .latest() on TimeSeries object

### PR DESCRIPTION
**Pull request by erlendvollset**

### CODE REVIEW

File: cognite/client/data_classes/time_series.py [hunk 1]
```diff
@@ -73,6 +73,44 @@ def plot(
...
+        """Returns the number of datapoints in this time series.
+
+        This result may not be completely accurate, as it is based on aggregates which may be occasionally out of date.
+
+        Returns:
+            int: The number of datapoints in this time series.
+        """
+        identifier = utils._auxiliary.assert_at_least_one_of_id_or_external_id(self.id, self.external_id)
+        dps = self._cognite_client.datapoints.retrieve(
+            start=0, end="now", aggregates=["count"], granularity="10d", **identifier
```
Comment (f1cognite) [inline cognite/client/data_classes/time_series.py]: Why 10?

File: cognite/client/data_classes/time_series.py [hunk 2]
```diff
@@ -73,6 +73,44 @@ def plot(
...
+        return None
+
+    def first(self) -> Optional["Datapoint"]:
+        """Returns the first datapoint in this time series.
+
+        Returns:
+            Datapoint: A datapoint object containing the value and timestamp of the first datapoint.
+        """
+        identifier = utils._auxiliary.assert_at_least_one_of_id_or_external_id(self.id, self.external_id)
+        dps = self._cognite_client.datapoints.retrieve(**identifier, start=0, end="now", limit=1)
```
Comment (f1cognite) [inline cognite/client/data_classes/time_series.py]: Are you guaranteed that they are sorted by time 🤔 ?

Comment (erlendvollset) [reply cognite/client/data_classes/time_series.py]: Yes.

File: cognite/client/data_classes/time_series.py (hunk 1 above)
Comment (erlendvollset) [reply cognite/client/data_classes/time_series.py]: I guess I just had to pick something. 10d * 10000 dps limit = 273 years, so then we will always get the counts in a single request.

**Pull request merged**